crates/router/src/utils/verification.rs

```diff
@@ -151,30 +145,34 @@ async fn check_existence_and_add_domain_to_db(
         .collect();
 
     already_verified_domains.append(&mut new_verified_domains);
-
-    let update_business_profile = business_profile::BusinessProfileUpdateInternal {
-        applepay_verified_domains: Some(already_verified_domains),
-        profile_name: Some(business_profile.profile_name),
-        modified_at: Some(business_profile.modified_at),
-        return_url: business_profile.return_url,
-        enable_payment_response_hash: Some(business_profile.enable_payment_response_hash),
-        payment_response_hash_key: business_profile.payment_response_hash_key,
-        redirect_to_merchant_with_http_post: Some(
-            business_profile.redirect_to_merchant_with_http_post,
-        ),
-        webhook_details: business_profile.webhook_details,
-        metadata: business_profile.metadata,
-        routing_algorithm: business_profile.routing_algorithm,
-        intent_fulfillment_time: business_profile.intent_fulfillment_time,
-        frm_routing_algorithm: business_profile.frm_routing_algorithm,
-        payout_routing_algorithm: business_profile.payout_routing_algorithm,
-        is_recon_enabled: Some(business_profile.is_recon_enabled),
+    let updated_mca = storage::MerchantConnectorAccountUpdate::Update {
+        merchant_id: None,
```

Comment:
Create a separate enum variant which has `applepay_verified_domains` field

Reply:
We will refactor it back to `business_profile` again once it will be used by dashboard as discussed.